PINCache/PINOperationQueue.m

```diff
@@ -25,12 +25,14 @@ @interface PINOperationQueue () {
   dispatch_queue_t _concurrentQueue;
   dispatch_queue_t _semaphoreQueue;
   
-  NSMutableOrderedSet *_allOperations;
+  NSMutableOrderedSet *_queuedOperations;
   NSMutableOrderedSet *_lowPriorityOperations;
   NSMutableOrderedSet *_defaultPriorityOperations;
   NSMutableOrderedSet *_highPriorityOperations;
   
-  NSHashTable *_canceledOperations;
+  NSMapTable *_referenceToOperations;
+  
+  NSArray <NSMutableOrderedSet *> *_prioritizedOperationQueues;
```

Comment:
I might consider just using a C array here as a micro-optimization, sized to `PINOperationQueuePriorityHigh`. It doesn't need to retain any of its items because their lifetimes are already bound to this class' lifetime.